dali/operators/geometry/coord_transform.h

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GEOMETRY_COORD_TRANSFORM_H_
+#define DALI_OPERATORS_GEOMETRY_COORD_TRANSFORM_H_
+
+#include <vector>
+#include "dali/core/format.h"
+#include "dali/core/geom/mat.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/operators/geometry/mt_transform_attr.h"
+
+namespace dali {
+
+#define COORD_TRANSFORM_INPUT_TYPES (uint8_t, int16_t, uint16_t, int32_t, float)
```

Comment:
We need to limit the number of types, because the dimensionality is handled at compile-time and we already have 36 combinations there.
This set of types will allow us to process:
- colors (uint8, uint16)
- audio (int16, int32)
- geometry (float)